### PR TITLE
add no-default-lib tag to core libraries

### LINKED ISSUE #6442: Performance is 3x slower in typescript@next
Apologies if you are already aware of this. When running my unit tests for [plugin-typescript](https://github.com/frankwallis/plugin-typescript) against typescript@next I am seeing a ~3x performance slowdown when type-checking:

```
   git clone https://github.com/frankwallis/plugin-typescript.git
   cd plugin-typescript
   npm i
   npm test

   [see output 1]

   npm install typescript@next
   npm test

   [see output 2]
```
output 1:
```
  Type Checker ES6
    ✓ compiles successfully (164ms)
    ✓ loads lib.d.ts by default (145ms)
    ✓ uses config options (163ms)
    ✓ compiles ambient imports (150ms)
    ✓ catches type errors (174ms)
    ✓ catches nested type-checker errors (169ms)
    ✓ fetches all the files needed for compilation (196ms)
    ✓ catches syntax errors (169ms)
    ✓ catches syntax errors in reference files (171ms)
    ✓ handles ambient references when resolveAmbientRefs option is false (240ms)
    ✓ resolves ambient references when resolveAmbientRefs option is true (265ms)
    ✓ handles ambient javascript imports (179ms)
    ✓ handles circular references (167ms)
    ✓ handles ambient typescript imports (170ms)
    ✓ resolves ambient typescript imports (157ms)
    ✓ handles ambients with subset names (164ms)
    ✓ handles ambients with internal requires (150ms)
    ✓ handles external imports (155ms)
    ✓ imports .css files (155ms)
    ✓ imports .html files (168ms)
    ✓ resolves typings files from package.json when resolveTypings is true (187ms)
    ✓ doesnt resolve typings files when resolveTypings is false (163ms)
    ✓ handles missing typings field in package.json (186ms)
    ✓ handles non-relative typings field in package.json (168ms)
TypeScript unable to resolve typings for missing_package, missing_package/package.json could not be found
    ✓ handles package.json not found (164ms)
```

output 2:
```
  Type Checker ES6
    ✓ compiles successfully (639ms)
    ✓ loads lib.d.ts by default (572ms)
    ✓ uses config options (590ms)
    ✓ compiles ambient imports (586ms)
    ✓ catches type errors (549ms)
    ✓ catches nested type-checker errors (643ms)
    ✓ fetches all the files needed for compilation (760ms)
    ✓ catches syntax errors (533ms)
    ✓ catches syntax errors in reference files (452ms)
    ✓ handles ambient references when resolveAmbientRefs option is false (476ms)
    ✓ resolves ambient references when resolveAmbientRefs option is true (459ms)
    ✓ handles ambient javascript imports (470ms)
    ✓ handles circular references (483ms)
    ✓ handles ambient typescript imports (493ms)
    ✓ resolves ambient typescript imports (481ms)
    ✓ handles ambients with subset names (479ms)
    ✓ handles ambients with internal requires (496ms)
    ✓ handles external imports (475ms)
    ✓ imports .css files (485ms)
    ✓ imports .html files (474ms)
    ✓ resolves typings files from package.json when resolveTypings is true (505ms)
    ✓ doesnt resolve typings files when resolveTypings is false (735ms)
    ✓ handles missing typings field in package.json (547ms)
    ✓ handles non-relative typings field in package.json (497ms)
TypeScript unable to resolve typings for missing_package, missing_package/package.json could not be found
    ✓ handles package.json not found (436ms)
```

### PR DESCRIPTION
**Pull request by vladima**
fixes #6442

### CODE REVIEW

File: src/lib/core.d.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
```
Comment (DanielRosenwasser) [inline src/lib/core.d.ts]: Shouldn't this still be here?

Comment (vladima) [inline src/lib/core.d.ts]: no, all core libraries get this line from `header.d.ts`

Comment (DanielRosenwasser) [inline src/lib/core.d.ts]: Oh sorry I thought this was `lib/core.d.ts`

**Comment by DanielRosenwasser** [comment]
:+1: :+1: :+1: :+1:

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**